### ISSUE TITLE
Multi-Currency

### ISSUE THREAD

**Issue by stefanhopman1** [domits1/Domits#612]
## Feature Summary

Build a simple **Multi-Currency** capability for Hosts and Property Managers that supports:

* [ ] US Dollar (USD)
* [ ] Euro (EUR)

The goal of Phase 1 is to allow Hosts and Property Managers to operate their Domits properties and pricing in either USD or EUR.

### Supported Currencies

* [ ] **US Dollar (USD)** — Global standard and fallback currency.
* [ ] **Euro (EUR)** — Used across the Eurozone in the EU.

Do not build full enterprise multi-currency functionality in this phase.

---

## Problem Statement

Hosts and Property Managers operate properties in different markets.

Some properties need to be priced and managed in EUR, while others may use USD.

Without basic currency support, users may have to manually convert prices or use an unsuitable default currency.

This can create:

* [ ] Incorrect property pricing
* [ ] Confusing guest-facing prices
* [ ] Manual currency conversion
* [ ] Inconsistent reservation values
* [ ] Confusing financial information

The first phase should solve the core problem without introducing unnecessary complexity.

---

## Desired Outcome

A Host or Property Manager should be able to select **USD or EUR** as the currency for their property.

Once selected:

```text
Property
   ↓
Currency
   ↓
USD or EUR
   ↓
Rates
   ↓
Fees
   ↓
Reservations
   ↓
Guest-facing prices
```

The selected property currency should be consistently used throughout the relevant PMS flows.

---

## Target Users

* [ ] Guests / Customers
* [x] Hosts / Property Managers / Enterprise
* [x] Admin / Internal users
* [x] Developers
* [ ] Other: _______

### Primary Users

* Hosts managing 1–3 properties
* Small Property Managers managing multiple properties
* Domits internal administrators

---

### Impact level

* [ ] Low (nice-to-have)
* [x] Medium (improves workflows)
* [ ] High (unblocks key use cases)
* [ ] Critical (strategic / revenue-impacting)

### User reach

* [ ] Few users
* [x] Many users
* [ ] All users
* [ ] Unknown

---

### 1. Currency Selection

Add a simple currency setting to the Host / Property settings.

Available options:

```text
Currency

○ EUR — Euro
○ USD — US Dollar
```

The selected currency becomes the property's primary currency.

---

### 2. Default Currency

USD should be the system fallback currency.

If no currency has been explicitly configured:

```text
Default Currency = USD
```

For existing properties, the current system behavior should remain unchanged.

---

### 3. Property Currency

Each property should have one primary currency.

Example:

```text
Villa Amsterdam
Currency: EUR

Villa Miami
Currency: USD
```

A Host can therefore manage properties using different currencies.

---

### 4. Pricing

All property pricing should use the property's configured currency.

This includes:

* [ ] Nightly price
* [ ] Weekly price
* [ ] Monthly price
* [ ] Minimum-stay pricing
* [ ] Discounts
* [ ] Additional fees
* [ ] Cleaning fees
* [ ] Taxes where applicable
* [ ] Rate plans

Example:

```text
Villa Amsterdam

€350 / night
€100 cleaning fee
```

or:

```text
Villa Miami

$450 / night
$120 cleaning fee
```

---

### 5. Reservations

Reservations should use the property's configured currency.

Example:

```text
Property Currency: EUR

Reservation
€2,450
```

or:

```text
Property Currency: USD

Reservation
$3,200
```

The reservation should store the currency together with the monetary amount.

---

### 6. Guest-Facing Prices

Guests should clearly see the property's currency.

Examples:

```text
€350 / night
```

```text
$450 / night
```

The currency symbol and/or ISO currency code should be displayed consistently.

---

### 7. Host Dashboard

Hosts should be able to see the currency throughout the PMS.

Examples:

* [ ] Calendar
* [ ] Property rates
* [ ] Reservations
* [ ] Reservation details
* [ ] Fees
* [ ] Revenue
* [ ] Reports
* [ ] Payout information where applicable

---

### Currency Setup

* [ ] As a Host, I want to select EUR as my property currency so I can manage European properties in euros.
* [ ] As a Host, I want to select USD as my property currency so I can manage properties priced in US dollars.
* [ ] As a Property Manager, I want to select a currency for each property so I can manage properties in different markets.
* [ ] As a Host, I want USD to be used when no currency has been configured so my property always has a valid currency.

### Pricing

* [ ] As a Host, I want to enter nightly rates in my property's currency so I don't have to manually convert prices.
* [ ] As a Host, I want additional fees to use my property's currency so pricing remains consistent.
* [ ] As a Host, I want discounts to be calculated in my property's currency.
* [ ] As a Host, I want the calendar to display prices using my property's currency.

### Reservations

* [ ] As a Host, I want reservations to display the property's currency so I immediately understand the booking value.
* [ ] As a Property Manager, I want each reservation to store its currency so historical reservation values remain clear.
* [ ] As a Host, I want reservation totals to use the property's currency.

### Guest Experience

* [ ] As a Guest, I want to clearly see whether a price is in USD or EUR so I understand the booking cost.
* [ ] As a Guest, I want pricing to remain consistent throughout the booking process.

---

### Phase 1

Only these currencies are supported:

| Currency  | Code | Default |
| --------- | ---- | ------- |
| US Dollar | USD  | **Yes** |
| Euro      | EUR  | No      |

### Rules

* [ ] Every property must have one currency.
* [ ] Currency must be either USD or EUR.
* [ ] USD is the fallback currency.
* [ ] Pricing uses the property's currency.
* [ ] Reservations store the property's currency.
* [ ] Existing transactions must retain their original currency.
* [ ] Currency should not be silently converted.
* [ ] Unsupported currencies should not be accepted.

---

## Changing Property Currency

Changing the property's currency should be treated carefully.

Example:

```text
Current:
EUR

Change to:
USD
```

The system should:

* [ ] Require appropriate permission.
* [ ] Clearly warn the user.
* [ ] Apply the new currency only to future pricing where appropriate.
* [ ] Preserve historical reservation currencies.
* [ ] Never silently convert historical financial records.

---

## Alternatives Considered

* [ ] USD only
* [ ] EUR only
* [ ] Automatically convert all prices
* [ ] Allow users to manually enter currency symbols
* [ ] Support many currencies immediately

The recommended approach is **USD + EUR only** for Phase 1.

---

### Performance

* [ ] Currency selection should not negatively affect PMS performance.
* [ ] No external exchange-rate API is required for basic Phase 1 functionality.

### Security

* [ ] Only authorized users can change property currency.
* [ ] Currency changes should be logged where appropriate.

### Privacy

* [ ] No additional personal data is required.

### Accessibility

* [ ] Currency must not be communicated only through color.
* [ ] Currency code should be available where the symbol could be ambiguous.

### Backward Compatibility

* [ ] Existing properties should continue functioning.
* [ ] Existing pricing should not unexpectedly change.
* [ ] Existing reservations should retain their existing currency.
* [ ] Existing APIs should remain compatible where possible.

### External Dependencies

Phase 1 should **not require an external exchange-rate provider**.

---

### Property Currency

* [ ] Given a Host creates a property, when no currency is selected, then USD is used as the default.

* [ ] Given a Host selects EUR, when the property is saved, then the property currency is EUR.

* [ ] Given a Host selects USD, when the property is saved, then the property currency is USD.

### Pricing

* [ ] Given a property uses EUR, when the Host enters a nightly price of 350, then Domits displays €350.

* [ ] Given a property uses USD, when the Host enters a nightly price of 450, then Domits displays $450.

### Reservations

* [ ] Given a property uses EUR, when a reservation is created, then the reservation amount is stored in EUR.

* [ ] Given a property uses USD, when a reservation is created, then the reservation amount is stored in USD.

### Historical Data

* [ ] Given a reservation was created in EUR, when the property currency later changes to USD, then the historical reservation remains recorded in EUR.

### Unsupported Currencies

* [ ] Given a user attempts to configure a currency other than USD or EUR, then Domits prevents the selection.

---

### Simple Currency Architecture

```text
              HOST / PROPERTY MANAGER
                       │
                       ▼
                    PROPERTY
                       │
                ┌──────┴──────┐
                │             │
               USD           EUR
                │             │
                ▼             ▼
             Pricing       Pricing
                │             │
                ▼             ▼
           Reservations  Reservations
                │             │
                ▼             ▼
             Guest         Guest
            Display       Display
```

### Example

```text
Host
│
├── Villa Amsterdam
│   └── EUR
│
├── Villa Barcelona
│   └── EUR
│
└── Villa Miami
    └── USD
```

---

## Future Phases

Phase 1 should create the foundation for future multi-currency capabilities.

### Phase 2

* [ ] GBP
* [ ] CHF
* [ ] CAD
* [ ] AUD
* [ ] Other major currencies
* [ ] Guest display currency
* [ ] Basic exchange-rate conversion

### Phase 3

* [ ] Enterprise reporting currency
* [ ] Multi-currency financial reporting
* [ ] Historical exchange rates
* [ ] Multi-currency payouts
* [ ] Multi-currency accounting
* [ ] Currency conversion analytics

### Phase 4

* [ ] Full enterprise multi-currency
* [ ] Legal-entity currencies
* [ ] Portfolio currencies
* [ ] Automated FX management
* [ ] Advanced reconciliation
* [ ] FX gain/loss reporting

----

**Key Currencies for Domits**
Selecting a range of commonly used currencies helps accommodate users from diverse economic regions and allows for easy local transactions.

**First phase**
- [ ] US Dollar (USD) – Global standard and often the fallback currency.
- [ ] Euro (EUR) – Used across the Eurozone in the EU.

**Next phase**
- [ ] British Pound (GBP) – Essential for the U.K.
- [ ] Swiss Franc (CHF) – Switzerland and parts of Europe.
- [ ] Turkish Lira (TRY) – For Turkey and surrounding areas.

**Next phase**
- [ ] Japanese Yen (JPY) – Important for Japan.
- [ ] Chinese Yuan (CNY) – Necessary for users in mainland China.
- [ ] Indian Rupee (INR) – Important for India, a large emerging market.
- [ ] Australian Dollar (AUD) – For Australia and the Oceania region.
- [ ] Canadian Dollar (CAD) – For Canada.
- [ ] Brazilian Real (BRL) – Essential for Brazil and South America.
- [ ] Mexican Peso (MXN) – For Mexico, and significant in Latin America.
- [ ] Russian Ruble (RUB) – Important for Russia and some surrounding regions.
- [ ] South Korean Won (KRW) – Critical for South Korea’s highly active market.
- [ ] Singapore Dollar (SGD) – Key for Singapore and parts of Southeast Asia.
- [ ] Hong Kong Dollar (HKD) – Widely used in Hong Kong.
- [ ] Saudi Riyal (SAR) – For Saudi Arabia and parts of the Gulf.
- [ ] South African Rand (ZAR) – For South Africa and some African regions.
- [ ] Indonesian Rupiah (IDR) – For Indonesia, a rapidly growing market.

**Comment by stefanhopman1** [comment]
User Story 1: Multi-Language Support
As a global user,
I want to view the platform in my preferred language,
so that I can easily understand and navigate the platform.

What If’s
What if the user’s preferred language is not supported?

Solution: Show the default language (e.g., English) and allow users to choose from supported languages.
What if the user’s browser language and their IP-based location suggest different languages?

Solution: Use the browser’s language setting as the primary reference and IP-based location as a secondary fallback.
What if the user changes their language preference mid-session?

Solution: Apply the language change instantly, with all displayed content updated without needing a page reload.
What if new content is added without translation?

Solution: Display new content in the default language and prioritize translations in the backend for updates.
What if certain phrases or industry terms do not translate well?

Solution: Use localized terminology experts or region-specific phrasing to adapt translations as needed.
Acceptance Criteria
Language Selection Option:

GIVEN the user accesses language settings,
WHEN they choose a language,
THEN the entire platform’s content should display in the selected language.
Browser Language Detection:

GIVEN no language preference is set,
WHEN the user accesses the platform,
THEN the platform should default to the browser’s language if supported, otherwise, fall back to the default language.
Real-Time Language Switching:

GIVEN the user is viewing content on the platform,
WHEN they switch languages mid-session,
THEN all content should instantly display in the selected language without requiring a page reload.
Fallback for Missing Translations:

GIVEN a specific piece of content lacks translation,
WHEN the user views that content,
THEN it should display in the platform’s default language.
User Story 2: Multi-Currency Support
As a global user,
I want to see prices in my preferred currency,
so that I understand costs without needing to convert them manually.

What If’s
What if the user’s currency is unavailable?

Solution: Default to the primary currency (e.g., USD) and offer a currency selector for the user to choose an available option.
What if the exchange rate fluctuates frequently?

Solution: Set up daily or hourly updates for exchange rates and cache them to reduce frequent API calls.
What if the user prefers a specific currency that doesn’t match their location?

Solution: Allow users to manually select their preferred currency from a list.
What if prices are displayed as uneven decimal numbers (e.g., $4.37)?

Solution: Apply rounding rules for each currency to make prices more user-friendly (e.g., $4.99).
What if prices need to display localized pricing rather than direct conversions?

Solution: Support localized pricing where possible and display set regional prices instead of exchange-based values for those areas.
Acceptance Criteria
Currency Detection by Location:

GIVEN the user has not set a preferred currency,
WHEN they access the platform,
THEN the platform should detect their location and display prices in the local currency if supported.
Manual Currency Selection:

GIVEN the user accesses the currency settings,
WHEN they select a currency from the list,
THEN all displayed prices should reflect the chosen currency.
Exchange Rate Update Handling:

GIVEN the currency rate has changed,
WHEN the user views the product prices,
THEN prices should reflect the latest exchange rate within the last update interval (e.g., daily or hourly).
Localized Pricing Display:

GIVEN a region has predefined prices,
WHEN a user from that region views the prices,
THEN display the localized price instead of a converted value.
Rounding Prices:

GIVEN prices are displayed in a selected currency,
WHEN the prices involve conversions,
THEN the amounts should be rounded according to the local currency standards (e.g., nearest 0.99).
User Story 3: Persistent Language and Currency Preferences
As a returning user,
I want my language and currency preferences saved,
so that I do not have to set them each time I log in.

What If’s
What if the user logs in from a new device?

Solution: Pull the user’s saved preferences from their profile and apply them on any device they log into.
What if the user wants different settings on different devices?

Solution: Allow device-based session preferences for language and currency as a secondary setting option.
What if a guest user (not logged in) selects preferences?

Solution: Save preferences temporarily for the session, but do not persist across different sessions.
What if the user updates their preferences mid-session?

Solution: Immediately apply updated preferences for language and currency across all relevant content.
What if a preference cannot be saved due to a network issue?

Solution: Display a message indicating that preferences couldn’t be saved and retry once the connection is restored.
Acceptance Criteria
Load Saved Preferences on Login:

GIVEN a returning user with saved preferences,
WHEN they log in,
THEN the platform should apply saved language and currency settings automatically.
Temporary Preferences for Guest Users:

GIVEN a guest user sets language and currency preferences,
WHEN they browse the platform during that session,
THEN the preferences should be applied for the current session only, without saving for future visits.
Real-Time Preference Update:

GIVEN the user is viewing content,
WHEN they change language or currency preferences,
THEN the platform should apply these changes immediately.
Error Handling for Preference Saving:

GIVEN there’s a network or system error,
WHEN a user tries to save their preferences,
THEN an error message should display, and the system should retry the save once the network is restored.
